lib/block-directory.php

```diff
@@ -0,0 +1,22 @@
+<?php
+/**
+ * Block directory functions.
+ *
+ * @package gutenberg
+ */
+
+if ( ! has_action( 'admin_enqueue_scripts', 'gutenberg_aenqueue_block_editor_assets_block_directory' ) ) {
+	/**
+	 * Function responsible for enqueuing the assets required
+	 * for the block directory functionality in the editor.
+	 */
+	function gutenberg_enqueue_block_editor_assets_block_directory() {
+		if ( ! gutenberg_is_experiment_enabled( 'gutenberg-block-directory' ) ) {
+			return;
+		}
+
+		wp_enqueue_script( 'wp-block-directory' );
+		wp_enqueue_style( 'wp-block-directory' );
+	}
+	add_action( 'enqueue_block_editor_assets', 'gutenberg_enqueue_block_editor_assets_block_directory' );
```

Comment:
Would it work if we only add the action if the gutenberg-block-directory experiment is enabled instead of checking inside the action?

Reply:
It works either way, I mirrored the logic from other actions, e.g.:

https://github.com/WordPress/gutenberg/blob/4c53811433363368a99e2db3f33a0dd4ec1ea3c7/lib/templates.php#L11-L17

I can update if you think it makes any difference.

Reply:
Having the check before adding the action would allow third-party code to check if the action is going to execute or not by using `has_action( 'enqueue_block_editor_assets', 'gutenberg_enqueue_block_editor_assets_block_directory'... )´. If we do the check inside the action has_action will always return true even if the action does nothing.
But if the code we already have checks inside the action, I guess it makes sense to do the same. It is something we can bulk change later.

Reply:
Right, I will update tomorrow, in fact we can guard the part which loads this file. I had it implemented this way at some point. What do you think?

Reply:
Moved in 77d80f1.